vyper/builtins/_signatures.py

```diff
@@ -104,7 +103,9 @@ def _validate_arg_types(self, node):
 
         for kwarg in node.keywords:
             kwarg_settings = self._kwargs[kwarg.arg]
-            if kwarg_settings.require_literal and not isinstance(kwarg.value, vy_ast.Constant):
+            is_literal_value = prefold(kwarg.value) is not None
```

Comment:
this looks like a potential performance problem, since it could be called every time the AST is traversed